presto-tablestore/src/main/java/com/facebook/presto/tablestore/TablestoreFacade.java

```diff
@@ -0,0 +1,2003 @@
+    {
+        return ImmutableList.of(config.getInstance());
+    }
+
+    public List<SchemaTableName> listTables(ConnectorSession session, @Nullable String schemaNameOrNull)
+    {
+        ListTableResponse listTableResponse = syncClient.listTable();
+        return listTableResponse.getTableNames().stream()
+                .map(x -> new SchemaTableName(config.getInstance(), x))
+                .collect(toList());
+    }
+
+    public List<? extends ConnectorSplit> getSplits(ConnectorSession session, TablestoreTableLayoutHandle ol)
+    {
+        String stn = ol.getTable().getPrintableStn();
+        String queryId = session.getQueryId();
+        String ramUserId = session.getUser();
+        try {
+            log.info("getSplits() start, queryId=%s ramUserId=%s table=%s", queryId, ramUserId, stn);
+            TablestoreTableHandle th = ol.getTable();
+            TupleDomain<ColumnHandle> constraint = ol.getTupleDomain().orElseThrow(NullPointerException::new);
+            if (constraint.isNone()) {
+                return emptyList();
+            }
+
+            Map<ColumnHandle, Domain> domains = constraint.getDomains().orElseThrow(NullPointerException::new);
+            Optional<ColumnHandle> pkIsNull = domains.entrySet().stream().filter(k -> {
+                TablestoreColumnHandle kk = (TablestoreColumnHandle) k.getKey();
+                Domain dv = k.getValue();
+                return kk.isPrimaryKey() && dv.isOnlyNull();
+            }).findFirst().map(Entry::getKey);
+
+            String tupleDomainStr = constraint.toString(session.getSqlFunctionProperties());
+            if (pkIsNull.isPresent()) {
+                log.info("The nullable value of primary key[%s] isn't allowed in Tablestore any time, " +
+                                "so we ignores this tuple domain, queryId=%s tupleDomain=%s",
+                        ((TablestoreColumnHandle) pkIsNull.get()).getColumnName(), queryId, tupleDomainStr);
+                return emptyList();
+            }
+
+            Map<String, Set<String>> matchedSearchIndexes = findMatchedSearchIndex(session, ol);
+
+            List<TablestoreSplit> x;
+            long t1 = currentTimeMillis();
+            if (matchedSearchIndexes.size() > 0) {
+                boolean isParallelScanMode = isParallelScanMode(session);
+                log.info("query %s is using ParallelScanMode: %s, matched search index: %s", queryId, isParallelScanMode, matchedSearchIndexes);
+
+                if (isParallelScanMode) {
+                    x = computeSplitsFromSearchIndex(session, ol, constraint, tupleDomainStr, matchedSearchIndexes);
+                }
+                else {
+                    x = querySplitsFromSearchIndex(session, th, constraint, tupleDomainStr, matchedSearchIndexes);
+                    t1 = currentTimeMillis() - t1;
+                    log.info("getSplits() end, found %s indexes, query=%s splitsSize=%s apiCost=%sms.",
```

Comment:
Will this generate too many logs?  

If the splits size and t1 are important can we expose them by jmx?

Reply:
it only generate one log per query.